### PR TITLE
[Core][Channel] Refactor migrate and broadcast channel

### PR DESCRIPTION
**Pull request by ClydeZhao**

### CODE REVIEW

File: core/channel/migrate_channel.hpp [hunk 1]
```diff
@@ -29,40 +29,27 @@ namespace husky {
...
     void migrate(ObjT& obj, int dst_thread_id) {
-        auto idx = this->src_ptr_->delete_object(&obj);
+        auto idx = this->obj_list_ptr_->delete_object(&obj);
         migrate_buffer_[dst_thread_id] << obj;
-        this->src_ptr_->migrate_attribute(migrate_buffer_[dst_thread_id], idx);
+        this->obj_list_ptr_->migrate_attribute(migrate_buffer_[dst_thread_id], idx);
     }
 
-    void prepare() override {}
+    void set_obj_list(ObjList<ObjT>* obj_list_ptr) { obj_list_ptr_ = obj_list_ptr; }
```
Comment (ddmbr) [inline core/channel/migrate_channel.hpp:50]: How about we make two methods instead: `set_src_obj_list` and `set_dst_obj_list`?

Comment (ClydeZhao) [reply core/channel/migrate_channel.hpp:50]: The channel does not depend on `dst_list` to `send`. So I think there is no need for `set_dst_obj_list`. Anyway, changing the name to `set_src_obj_list` makes the code clearer.

Comment (ddmbr) [reply core/channel/migrate_channel.hpp:50]: I see

**Comment by ddmbr** [comment]
I guess this PR is now ready?

**Comment by ClydeZhao** [comment]
Yes, it passed the test cases.

2017-02-08 14:13 GMT+08:00 YANG Fan <notifications@github.com>:

**Pull request merged**